From 36e738e11a9b745a2fb4e8f3cd0ad1d511d9bf5c Mon Sep 17 00:00:00 2001
From: Lars Ellenberg <lars.ellenberg@linbit.com>
Date: Mon, 9 Apr 2018 16:56:36 +0200
Subject: [PATCH] fix name passed to dlopen: libsqlite3.so.0

the libsqlite3.so symlink is only present with the -dev package installed :-(
---
 db_sqlite.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/db_sqlite.c b/db_sqlite.c
index 45d6725..0f673bf 100644
--- a/db_sqlite.c
+++ b/db_sqlite.c
@@ -56,16 +56,16 @@ static void *dl_handle;
 
 static void db_sqlite3_dlopen(void)
 {
-	csync_debug(2, "Opening shared library libsqlite3.so\n");
+	csync_debug(2, "Opening shared library libsqlite3.so.0\n");
 
-	dl_handle = dlopen("libsqlite3.so", RTLD_LAZY);
+	dl_handle = dlopen("libsqlite3.so.0", RTLD_LAZY);
 	if (dl_handle == NULL) {
 		csync_fatal
-		    ("Could not open libsqlite3.so: %s\n"
+		    ("Could not open libsqlite3.so.0: %s\n"
 		     "Please install sqlite3 client library (libsqlite3) or use other database (postgres, mysql)\n",
 		     dlerror());
 	}
-	csync_debug(2, "Reading symbols from shared library libsqlite3.so\n");
+	csync_debug(2, "Reading symbols from shared library libsqlite3.so.0\n");
 
 	LOOKUP_SYMBOL(dl_handle, sqlite3_open);
 	LOOKUP_SYMBOL(dl_handle, sqlite3_close);

